src/oncoming_example.cpp

```diff
@@ -0,0 +1,435 @@
+  // const auto p1_goaly_cost = std::make_shared<FinalTimeCost>(
+  //     std::make_shared<QuadraticCost>(kGoalCostWeight, kP1YIdx, kP1GoalY),
+  //     kTimeHorizon - kFinalTimeWindow, "GoalY");
+  // p1_cost.AddStateCost(p1_goalx_cost);
+  // p1_cost.AddStateCost(p1_goaly_cost);
+
+  // const auto p2_goalx_cost = std::make_shared<FinalTimeCost>(
+  //     std::make_shared<QuadraticCost>(kGoalCostWeight, kP2XIdx, kP2GoalX),
+  //     kTimeHorizon - kFinalTimeWindow, "GoalX");
+  // const auto p2_goaly_cost = std::make_shared<FinalTimeCost>(
+  //     std::make_shared<QuadraticCost>(kGoalCostWeight, kP2YIdx, kP2GoalY),
+  //     kTimeHorizon - kFinalTimeWindow, "GoalY");
+  // p2_cost.AddStateCost(p2_goalx_cost);
+  // p2_cost.AddStateCost(p2_goaly_cost);
+
+  // const auto p3_goalx_cost = std::make_shared<FinalTimeCost>(
+  //     std::make_shared<QuadraticCost>(kGoalCostWeight, kP3XIdx, kP3GoalX),
+  //     kTimeHorizon - kFinalTimeWindow, "GoalX");
+  // const auto p3_goaly_cost = std::make_shared<FinalTimeCost>(
+  //     std::make_shared<QuadraticCost>(kGoalCostWeight, kP3YIdx, kP3GoalY),
+  //     kTimeHorizon - kFinalTimeWindow, "GoalY");
+  // p3_cost.AddStateCost(p3_goalx_cost);
+  // p3_cost.AddStateCost(p3_goaly_cost);
+
+  // Pairwise proximity costs.
+  const std::shared_ptr<ProxCost> p1p2_proximity_cost(
+      new ProxCost(kP1ProximityCostWeight, {kP1XIdx, kP1YIdx},
+                   {kP2XIdx, kP2YIdx}, kMinProximity, "ProximityP2"));
+  // const std::shared_ptr<ProxCost> p1p3_proximity_cost(
+  //    new ProxCost(kP1ProximityCostWeight, {kP1XIdx, kP1YIdx},
+  //                 {kP3XIdx, kP3YIdx}, kMinProximity, "ProximityP3"));
+  p1_cost.AddStateCost(p1p2_proximity_cost);
+  // p1_cost.AddStateCost(p1p3_proximity_cost);
+
+  const std::shared_ptr<ProxCost> p2p1_proximity_cost(
+      new ProxCost(kP2ProximityCostWeight, {kP2XIdx, kP2YIdx},
+                   {kP1XIdx, kP1YIdx}, kMinProximity, "ProximityP1"));
+  // const std::shared_ptr<ProxCost> p2p3_proximity_cost(
+  //     new ProxCost(kP2ProximityCostWeight, {kP2XIdx, kP2YIdx},
+                   // {kP3XIdx, kP3YIdx}, kMinProximity, "ProximityP3"));
+  p2_cost.AddStateCost(p2p1_proximity_cost);
+  //  p2_cost.AddStateCost(p2p3_proximity_cost);
+
+  // const std::shared_ptr<ProxCost> p3p1_proximity_cost(
+  //     new ProxCost(kP3ProximityCostWeight, {kP3XIdx, kP3YIdx},
+  //                  {kP1XIdx, kP1YIdx}, kMinProximity, "ProximityP1"));
+  // const std::shared_ptr<ProxCost> p3p2_proximity_cost(
+  //     new ProxCost(kP3ProximityCostWeight, {kP3XIdx, kP3YIdx},
+                   // {kP2XIdx, kP2YIdx}, kMinProximity, "ProximityP2"));
+  // p3_cost.AddStateCost(p3p1_proximity_cost);
+  // p3_cost.AddStateCost(p3p2_proximity_cost);
+
+  // Set up solver.
+  solver_.reset(new ILQSolver(dynamics, {p1_cost, p2_cost, p3_cost},
+                              kTimeHorizon, params));
+}
+
+inline std::vector<float> ThreePlayerOvertakingExample::Xs(
```

Comment:
gotta change the namespace here to be `OncomingExample` not `ThreePlayerOvertakingExample`, and likewise below. This is the source of your linker errors, I suspect.